src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -214,73 +217,88 @@ def _process_resource_change(
+                # we hit the max number of retries.
+                return retryer.retry(exc=e)
+
+        event = f"{name}.{action}"
+        lifecycle.add_extras(extras={"event_name": event, "instance_id": instance_id})
+
+        if event not in VALID_EVENTS:
+            lifecycle.record_failure(
+                failure_reason="invalid_event",
+            )
             return
-        instance = Event(
-            project_id=project_id, group_id=group_id, event_id=str(instance_id), data=nodedata
-        )
-        name = sender.lower()
-    else:
-        # Some resources are named differently than their model. eg. Group vs Issue.
-        # Looks up the human name for the model. Defaults to the model name.
-        name = RESOURCE_RENAMES.get(model.__name__, model.__name__.lower())
-
-    # By default, use Celery's `current_task` but allow a value to be passed for the
-    # bound Task.
-    retryer = retryer or current_task
-
-    # We may run into a race condition where this task executes before the
-    # transaction that creates the Group has committed.
-    if not issubclass(model, Event):
-        try:
-            instance = model.objects.get(id=instance_id)
-        except model.DoesNotExist as e:
-            # Explicitly requeue the task, so we don't report this to Sentry until
-            # we hit the max number of retries.
-            return retryer.retry(exc=e)
-
-    event = f"{name}.{action}"
 
-    if event not in VALID_EVENTS:
-        return
-
-    org = None
+        org = None
 
-    if isinstance(instance, (Group, Event, GroupEvent)):
-        org = Organization.objects.get_from_cache(
-            id=Project.objects.get_from_cache(id=instance.project_id).organization_id
-        )
-        assert org, "organization must exist to get related sentry app installations"
-
-        installations = [
-            installation
-            for installation in app_service.installations_for_organization(organization_id=org.id)
-            if event in installation.sentry_app.events
-        ]
-
-        for installation in installations:
-            data = {}
-            if isinstance(instance, (Event, GroupEvent)):
-                assert instance.group_id, "group id is required to create webhook event data"
-                data[name] = _webhook_event_data(instance, instance.group_id, instance.project_id)
-            else:
-                data[name] = serialize(instance)
-
-            # Trigger a new task for each webhook
-            send_resource_change_webhook.delay(
-                installation_id=installation.id, event=event, data=data
+        if isinstance(instance, (Group, Event, GroupEvent)):
+            org = Organization.objects.get_from_cache(
+                id=Project.objects.get_from_cache(id=instance.project_id).organization_id
             )
+            assert org, "organization must exist to get related sentry app installations"
```

Comment:
if this assert fails, it will cause the metric to fail and the task to be retried, is that the desired behavior?

Reply:
nah let's add `assertionerror` to our handling

Reply:
it's ok for the metric to fail?